File: core/src/main/java/com/larryhsiao/nyx/core/jots/moods/ConstRankedMood.java
package com.larryhsiao.nyx.core.jots.moods;

/**
 * Constant of {@link RankedMood}
 */
public class ConstRankedMood implements RankedMood {
    private final String mood;
    private final int usedTimes;

    public ConstRankedMood(String mood, int usedTimes) {
        this.mood = mood;
        this.usedTimes = usedTimes;
    }

    @Override
    public String mood() {
        return mood;
    }

    @Override
    public int usedTimes() {
        return usedTimes;
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/moods/ConstRankedMood.kt
package com.larryhsiao.nyx.core.jots.moods

/**
 * Constant of [RankedMood]
 */
class ConstRankedMood(private val mood: String?, private val usedTimes: Int) : RankedMood {
    override fun mood(): String? {
        return mood
    }

    override fun usedTimes(): Int {
        return usedTimes
    }
}